docs/espace/tutorials/upgradableContract/upgrade.md

```diff
@@ -0,0 +1,283 @@
+
+Next, we'll proceed with the tutorial on how to implement this pattern on Conflux eSpace using Hardhat.
+
+## 1. Project Setup
+
+First, ensure you have Node.js and npm installed. Then, create a new directory and initialize the project:
+
+```
+mkdir upgradeable-contract-demo
+cd upgradeable-contract-demo
+npm init -y
+npm install --save-dev hardhat @nomicfoundation/hardhat-toolbox dotenv
+```
+
+Next, initialize the Hardhat project and select the JavaScript default template:
+
+```
+npx hardhat
+```
+
+When prompted, choose "Create a JavaScript project". This will create a basic Hardhat project structure, including `contracts`, `scripts`, and `test` directories, as well as a default `hardhat.config.js` file.
+
+After completing these steps, you'll have a basic Hardhat project structure using JavaScript, ready for writing and deploying upgradeable contracts.
+
+## 2. Configure Hardhat
+
+Create a Hardhat configuration file:
+
+```javascript
+require("@nomicfoundation/hardhat-toolbox");
+require("dotenv").config();
+
+module.exports = {
+  solidity: "0.8.24",
+  networks: {
+    eSpaceTestnet: {
+      url: "https://evmtestnet.confluxrpc.com",
+      accounts: [process.env.PRIVATE_KEY],
+    },
+  },
+};
+```
+
+Create a .env file and add your private key:
+
+```
+PRIVATE_KEY=your_private_key_here
+```
+
+## 3. Write Smart Contracts
+
+Create a contracts directory and add the following contracts:
+
+```solidity
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.24;
+
+contract Logic1 {
+    // Address of the current implementation contract
+    address public implementation;
+    // Address of the admin who can upgrade the contract
+    address public admin;
+    // A string variable to demonstrate state changes
+    string public words;
+
+    // Function to set the 'words' variable
+    function foo() public {
+        words = "old";
+    }
+}
+```
+
+```solidity
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.24;
+
+contract Logic2 {
+    // Address of the current implementation contract
+    address public implementation;
+    // Address of the admin who can upgrade the contract
+    address public admin;
+    // A string variable to demonstrate state changes
+    string public words;
+
+    // Function to set the 'words' variable
+    // Note: This function is different from Logic1
+    function foo() public {
+        words = "new";
+    }
+}
+```
+
+## 4. Deployment Script
+
+Create a scripts directory and add the following script:
+
+```javascript
+const hre = require("hardhat");
+
+async function main() {
+  // Deploy Logic1 contract
+  const Logic1 = await hre.ethers.getContractFactory("Logic1");
+  const logic1 = await Logic1.deploy();
+  await logic1.waitForDeployment();
+  console.log("Logic1 deployed to:", await logic1.getAddress());
+
+  // Deploy SimpleUpgrade (Proxy) contract
+  const SimpleUpgrade = await hre.ethers.getContractFactory("SimpleUpgrade");
```

Comment:
There is no code specified for this smart contract. The smart contract is missing so this deploy script doesn't work and show this error message:
`hardhaterror: HH700: Artifact for contract "SimpleUpgrade" not found.`

Please add the SimpleUpgrade code.